lsp.go

```diff
@@ -346,26 +363,49 @@ func checkFile(fname string, content string) error {
 		if strings.HasSuffix(filename, ".tm") || strings.HasSuffix(filename, ".tm.hcl") {
 			path := filepath.Join(dir, filename)
 
-			if path == fname {
-				// file already added
+			if path == fromFile {
+				// ignore source file
 				continue
 			}
 
-			contents, err := os.ReadFile(path)
-			if err != nil {
-				log.Error().Err(err).Send()
-				return err
-			}
+			files = append(files, path)
+		}
+	}
 
-			err = parser.AddFile(path, contents)
-			if err != nil {
-				log.Error().Err(err).Send()
-				return err
-			}
+	return files, nil
+}
+
+// checkFiles checks if the given provided files have errors but the thisFile
+// is handled separately because it can be unsaved.
+func checkFiles(files []string, thisFile string, thisContent string) error {
+	dir := filepath.Dir(thisFile)
+	parser := hcl.NewTerramateParser(dir)
+
+	for _, fname := range files {
+		var (
+			contents []byte
+			err      error
+		)
+
+		if thisFile == fname {
+			contents = []byte(thisContent)
+		} else {
+			contents, err = os.ReadFile(fname)
+		}
+
+		if err != nil {
+			log.Error().Err(err).Send()
+			return err
+		}
+
+		err = parser.AddFile(fname, contents)
+		if err != nil {
+			log.Error().Err(err).Send()
+			return err
 		}
 	}
 
 	log.Debug().Msg("about to parse all the files")
-	_, err = parser.Parse()
+	_, err := parser.Parse()
```

Comment:
If we need to log the errors here, it is missing here, although I think we could log on the caller if necessary.